### ISSUE TITLE
BV calculation for BNC-6S split location heavy gauss rifle looks incorrect

### ISSUE THREAD

**Issue by scottboehmer** [MegaMek/megameklab#881]
### Environment
What version of MekLab does your issue apply to? standalone MegaMekLab 0.46.1
What operating system are you using? Windows
What java version are you using? 1.8.0_281

### Description
The battle value calculation for the BNC-6S Banshee (using the built-in mech data) seems to have an error. MegaMekLab is deducting 11 points for the heavy gauss rifle, but 9 of the criticals for the weapon are located in the LT which is protected by CASE. It seems like only the 2 criticals in the CT should result in a defensive BV deduction.

I confirmed that if the entire heavy gauss is in a location protected by CASE then no points are deducted for its criticals, so it seems like this is an issue with the weapon being split between a protected location and an unprotected location.

Here is the calculation that I'm seeing in MegaMekLab:

```

Total Armor HD (9) x 1.0 | 9.0
-- | --
Total Armor CT (49) x 1.0 | 49.0
Total Armor RT (39) x 1.0 | 39.0
Total Armor LT (39) x 1.0 | 39.0
Total Armor RA (20) x 1.0 | 20.0
Total Armor LA (20) x 1.0 | 20.0
Total Armor RL (28) x 1.0 | 28.0
Total Armor LL (28) x 1.0 | 28.0
Total modified armor BV x 2.5 | = 580.0
Total I.S. Points x IS Multipler x 1.5 x Engine Multipler | 145 x 1.0 x 1.5 x 0.75 | = 163.125
Weight x Gyro Multipler | 95.0 x 0.5 | = 47.5
Defensive Equipment: |   |  
Total BV of all Defensive Equipment |   | = 0.0
Explosive Weapons/Equipment Penalty |   | = -11.0
  |   | -------------
  |   | 779.625
Run MP |   | 6
Target Movement Modifier For Run |   | 2
Target Movement Modifier For Jumping |   | 0
Target Movement Modifier For UMUs |   | 0
Target Movement Modifier |   | 2.0
Multiply by Defensive Movement Factor of | 1.2 | x 1.2
  |   | -------------
Defensive Battle Value |   | = 935.55
  |   |  
Offensive Battle Rating Calculation: |   |  
Base Heat Efficiency - Run Heat | 26 - 2 | = 24
Unmodified Weapon BV: |   |  
Heavy Gauss Rifle |   | 346.0
LB 10-X AC |   | 148.0
Small Laser |   | 9.0
  |   | -------------
Unmodified Front BV: |   | 503.0
Unmodfied Rear BV: |   | 0.0
Unmodfied Turret BV: |   | 0.0
Total Unmodfied BV: |   | 503.0
Unmodified Front non-arm BV: |   | 503.0
Unmodfied Rear non-arm BV: |   | 0.0
Weapon Heat: |   |  
Heavy Gauss Rifle |   | + 2.0
LB 10-X AC |   | + 2.0
Small Laser |   | + 1.0
  |   | -------------
Total Heat: |   | = 5.0
Weapons with no heat at full BV:
Heat Modified Weapons BV: |   |  
Heavy Gauss Rifle |   | 346.0
LB 10-X AC |   | 148.0
Small Laser |   | 9.0
  |   | -------------
Total Weapons BV Adjusted For Heat: |   | 503.0
Misc Offensive Equipment: |   |  
Total Misc Offensive Equipment BV: |   | 0.0
Total Ammo BV: |   | 253.0
Add weight |   | +95.0
Final Speed Factor: |   | 1.12
Weapons BV * Speed Factor | 851.0 * 1.12 | = 953.1200000000001
Offensive BV + Defensive BV | 935.55 + 953.1200000000001 | = 1888.67
Total BV * Cockpit Modifier | 1888.67 * 1.0 | = 1889.0
  |   | -------------
Final BV |   | 1889.0
```

**Comment by HammerGS** [comment]
Please test this in the latest development branch. 46.1 is End of Life and the 47.17 Dev release is a release candidate for the new stable. I believe we have a fix for this in that release.

**Comment by neoancient** [comment]
This was fixed by MegaMek/megamek#2418

**Comment by scottboehmer** [comment]
I downloaded the 47.17 dev release and confirmed it's now fixed. Thanks!
